### PR TITLE
Add infra-diff script to generate patch between infras

### PR DESCRIPTION
**Pull request by flomonster**
The diff script generate a patch between two railjson file.

### CODE REVIEW

File: scripts/diff-infra.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    rjs_base = Path(argv[1])
+    rjs_cur = Path(argv[2])
+    if not rjs_base.is_file():
+        print(f"File not found '{rjs_base}'")
+        exit(1)
+    if not rjs_cur.is_file():
+        print(f"File not found '{rjs_cur}'")
+        exit(1)
+
+    with open(rjs_base) as f:
```
Comment (leovalais) [inline scripts/diff-infra.py]: I think you can also write `rjs_base.open()`

**Review by leovalais — APPROVED** [review]

File: scripts/diff-infra.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+def as_sets(obj):
+    """
+    Return a copy of the object but all inner lists are transform to sets.
+    """
+    res = obj
+    if isinstance(obj, dict):
+        res = {k: as_sets(v) for (k, v) in res.items()}
+    elif isinstance(obj, list):
+        if obj and isinstance(obj[0], dict):
+            print(obj)
```
Comment (Castavo) [inline scripts/diff-infra.py]: Is that voluntary ?

File: scripts/merge-infra.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    for (key, value) in rjs_a.items():
+        if type(value) != list:
+            continue
+        objects_a = {v["id"]: v for v in value}
+        for obj_b in rjs_b[key]:
+            obj_id = obj_b["id"]
+            # Duplicate key
+            if obj_id in objects_a:
+                if obj_b != objects_a[obj_id]:
```
Comment (Castavo) [inline scripts/merge-infra.py]: Why not use `as_sets` here too ?

File: scripts/diff-infra.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    "Route": "routes",
+    "SwitchType": "switch_types",
+    "Switch": "switches",
+    "TrackSectionLink": "track_section_links",
+    "TrackSection": "track_sections",
+    "SpeedSection": "speed_sections",
+    "Catenary": "catenaries",
+    "Signal": "signals",
+    "BufferStop": "buffer_stops",
+    "Detector": "detectors",
```
Comment (Castavo) [inline scripts/diff-infra.py:18]: This does not account for DeadSections, if this is voluntary it's okay

File: scripts/diff-infra.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    return {
+        "operation_type": "DELETE",
+        "obj_type": obj_type,
+        "obj_id": obj_id,
+    }
+
+
+def create_payload(obj_type: str, value: Dict) -> Dict:
+    """
+    Create the payload to indicate that an object must be deleted
```
Comment (Tguisnet) [inline scripts/diff-infra.py]:
```suggestion
    Create the payload to indicate that an object must be created
```

**Review by Tguisnet — APPROVED** [review]

**Review by leovalais — APPROVED** [review]
Still good to me

**Pull request merged**